### DISCUSSION TITLE
In reduce transforms, is there a limit on array concat merge strategy?

### DISCUSSION THREAD

**Discussion by ddayal219** [vectordotdev/vector#11494] [Q&A] [answered]
Currently, when you use an array concat merge strategy, I don't see a limit on the number of items you concatenate. The expire_after_ms is the only way I could find to indirectly control the the size of array, IE flush out with shorter expiry. 

I would like to check if there is any setting to explicitly set a limit on concatenated array and expire the state when we hit the array limit?

**Comment by spencergilbert** [comment, marked as answer]
Hi @ddayal219 - I don't think that's possible today, we do have this issue already open https://github.com/vectordotdev/vector/issues/9498 would that help in your usecase?
